src/software/amazon/ion/impl/IonReaderTextUserX.java

```diff
@@ -146,8 +146,11 @@ private final boolean has_next_user_value()
                         {
                             if (ION_1_0.equals(version))
                             {
-                                symbol_table_reset();
-                                push_symbol_table(_system_symtab);
+                                if (_value_keyword != IonTokenConstsX.KEYWORD_sid)
```

Comment:
I think it might make more sense to push this check into `isIonVersionMarker`, since it would apply to ANY version marker, not just ION_1_0.

Reply:
Let's keep it as-is.